cdap-app-fabric/src/test/java/co/cask/cdap/internal/AppFabricTestHelper.java

```diff
@@ -147,7 +148,11 @@ public static void ensureNamespaceExists(Id.Namespace namespace) throws Exceptio
   public static void ensureNamespaceExists(Id.Namespace namespace, CConfiguration cConf) throws Exception {
     NamespaceAdmin namespaceAdmin = getInjector(cConf).getInstance(NamespaceAdmin.class);
     if (!namespaceAdmin.exists(namespace)) {
-      namespaceAdmin.create(new NamespaceMeta.Builder().setName(namespace).build());
+      try {
+        namespaceAdmin.create(new NamespaceMeta.Builder().setName(namespace).build());
+      } catch (NamespaceAlreadyExistsException e) {
```

Comment:
no not a race condition. But the purpose of this method is to "ensure that a namespace exists". So if it already exists, it should not throw, but just ignore.